src/sentry/api/endpoints/rule_snooze.py

```diff
@@ -0,0 +1,118 @@
+import datetime
+
+from rest_framework import serializers, status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log, features
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectAlertRulePermission, ProjectEndpoint
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.api.serializers.rest_framework.base import CamelSnakeSerializer
+from sentry.incidents.models import AlertRule
+from sentry.models import Rule, RuleSnooze
+
+
+class RuleSnoozeValidator(CamelSnakeSerializer):
+    target = serializers.CharField(required=True, allow_null=False)
+    rule = serializers.BooleanField(required=False, allow_null=True)
+    alert_rule = serializers.BooleanField(required=False, allow_null=True)
+    until = serializers.DateTimeField(required=False, allow_null=True)
+
+
+@register(RuleSnooze)
+class RuleSnoozeSerializer(Serializer):  # type: ignore
+    def serialize(self, obj, attrs, user, **kwargs):
+        result = {
+            "ownerId": obj.owner_id,
+            "userId": obj.user_id or "everyone",
+            "until": obj.until or "forever",
+            "dateAdded": obj.date_added,
+            "ruleId": obj.rule_id,
+            "alertRuleId": obj.alert_rule_id,
+        }
+        return result
+
+
+def can_edit_alert_rule(rule, organization, user):
+    rule_owner = rule.owner
+    if rule_owner:
+        if rule_owner.team not in rule_owner.team.objects.get_for_user(organization, user):
```

Comment:
I'm not totally sure if the rule owner should be how we base the permissions. I think it should probably be based on the project that the rule is part of, and whether the user has access to that?